docs/guide/interchange/05-creating-sell-orders.md

```diff
@@ -0,0 +1,687 @@
+* Save the voucher received on the target chain to later resolve a denom
+* Transmit an IBC packet to the target chain
+
+```go
+// x/ibcdex/keeper/msg_server_sell_order.go
+import "errors"
+
+func (k msgServer) SendSellOrder(goCtx context.Context, msg *types.MsgSendSellOrder) (*types.MsgSendSellOrderResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	// If an order book doesn't exist, throw an error
+	pairIndex := types.OrderBookIndex(msg.Port, msg.ChannelID, msg.AmountDenom, msg.PriceDenom)
+	_, found := k.GetSellOrderBook(ctx, pairIndex)
+	if !found {
+		return &types.MsgSendSellOrderResponse{}, errors.New("the pair doesn't exist")
+	}
+	// Get sender's address
+	sender, err := sdk.AccAddressFromBech32(msg.Sender)
+	if err != nil {
+		return &types.MsgSendSellOrderResponse{}, err
+	}
+	// Use SafeBurn to ensure no new native tokens are minted
+	if err := k.SafeBurn(ctx, msg.Port, msg.ChannelID, sender, msg.AmountDenom, msg.Amount); err != nil {
+		return &types.MsgSendSellOrderResponse{}, err
+	}
+	// Save the voucher received on the other chain, to have the ability to resolve it into the original denom
+	k.SaveVoucherDenom(ctx, msg.Port, msg.ChannelID, msg.AmountDenom)
+	var packet types.SellOrderPacketData
+	packet.Seller = msg.Sender
+	packet.AmountDenom = msg.AmountDenom
+	packet.Amount = msg.Amount
+	packet.PriceDenom = msg.PriceDenom
+	packet.Price = msg.Price
+	// Transmit the packet
+	err = k.TransmitSellOrderPacket(ctx, packet, msg.Port, msg.ChannelID, clienttypes.ZeroHeight(), msg.TimeoutTimestamp)
+	if err != nil {
+		return nil, err
+	}
+	return &types.MsgSendSellOrderResponse{}, nil
+}
+```
+
+`SendSellOrder` depends on two new keeper methods: `SafeBurn` and `SaveVoucherDenom`.
+
+### `SafeBurn`
+
+`SafeBurn` burns tokens if they are IBC vouchers (have an `ibc/` prefix) and locks tokens if they are native to the chain.
+
+```go
+// x/ibcdex/keeper/mint.go
+package keeper
+
+import (
+  "fmt"
+  sdk "github.com/cosmos/cosmos-sdk/types"
+  ibctransfertypes "github.com/cosmos/cosmos-sdk/x/ibc/applications/transfer/types"
+  "github.com/username/interchange/x/ibcdex/types"
+  "strings"
+)
+
+// isIBCToken checks if the token came from the IBC module
+func isIBCToken(denom string) bool {
+  return strings.HasPrefix(denom, "ibc/")
+}
+
+func (k Keeper) SafeBurn(ctx sdk.Context, port string, channel string, sender sdk.AccAddress, denom string, amount int32) error {
+  if isIBCToken(denom) {
+    // Burn the tokens
+    if err := k.BurnTokens(ctx, sender, sdk.NewCoin(denom, sdk.NewInt(int64(amount)))); err != nil {
+      return err
+    }
+  } else {
+    // Lock the tokens
+    if err := k.LockTokens(ctx, port, channel, sender, sdk.NewCoin(denom, sdk.NewInt(int64(amount)))); err != nil {
+      return err
+    }
+  }
+  return nil
+}
+```
+
+Implement the `BurnTokens` keeper method.
+
+```go
+// x/ibcdex/keeper/mint.go
+func (k Keeper) BurnTokens(ctx sdk.Context, sender sdk.AccAddress, tokens sdk.Coin) error {
```

Comment:
Just realized that we do not burn Tokens. We burn vouchers but lock tokens, right? Maybe the functions can reflect this to avoid confusion.

Reply:
Good point. Let's do it in a separate PR, since this affects the code, to make sure nothing breaks in the tutorial.